lib/control-connection.js

```diff
@@ -439,8 +449,21 @@ class ControlConnection extends events.EventEmitter {
         hostIterator = await promiseUtils.newQueryPlan(this._profileManager.getDefaultLoadBalancing(), null, null);
       }
 
-      this.connection = this._borrowAConnection(hostIterator);
+      try {
 
+        this.connection = this._borrowAConnection(hostIterator);
+      }
+      catch(err) {
+
+        /* NODEJS-632: refresh nodes before getting hosts for reconnect since some hostnames may have
+         * shifted during the flight. */
+        this.log("info", "ControlConnection could not reconnect using existing connections.  Refreshing contact points and retrying");
+        await Promise.all(
+          this.options.contactPoints.map(name =>
+            this._clearResolvedContactPoints(name)
+            .then(this._parseContactPoint(name)))
+        ).then(this._initializeConnection());
```

Comment:
After digging into this a bit I suspect this is doing more work than we need to.  We should be okay just getting a new connection here... a full initialize might be a bit much.